File: 2016-07-12_rxjava_in_action/src/main/java/com/androidhuman/example/rxjavainaction/rx/RxContent.java
package com.androidhuman.example.rxjavainaction.rx;

import com.androidhuman.example.rxjavainaction.model.Content;
import com.androidhuman.example.rxjavainaction.model.User;

import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;

import rx.Observable;

public final class RxContent {

    @CheckResult
    @NonNull
    public static Observable<Void> checksAvailability(Content content) {
        return Observable.create(new ChecksAvailabilityOnSubscribe(content));
    }

    @CheckResult
    @NonNull
    public static Observable<Void> checksBalance(Content content, User user) {
        return Observable.create(new ChecksBalanceOnSubscribe(content, user));
    }

    @CheckResult
    @NonNull
    public static Observable<Void> checksContentRating(Content content, User user) {
        return Observable.create(new ChecksContentRatingOnSubscribe(content, user));
    }

}

File: 2016-07-12_rxjava_in_action/src/main/kotlin/com/androidhuman/example/rxjavainaction/rx/RxContent.kt
package com.androidhuman.example.rxjavainaction.rx

import com.androidhuman.example.rxjavainaction.model.Content
import com.androidhuman.example.rxjavainaction.model.User
import rx.Observable

public inline fun Content.checksAvailability(): Observable<Void>
        = RxContent.checksAvailability(this)

public inline fun Content.checksBalance(user: User): Observable<Void>
        = RxContent.checksBalance(this, user)

public inline fun Content.checksContentRating(user: User): Observable<Void>
        = RxContent.checksContentRating(this, user)
